spec/chrono_model/time_machine/history_spec.rb

```diff
@@ -54,6 +54,18 @@
       it { expect($t.foo.history.except(:order).select('max(id) as foo, min(id) as bar').group('id').first.attributes.keys).to match_array %w[id foo bar] }
     end
 
+    context 'when finding historical elements by hid' do
+      subject(:historical_foo) { Foo::History.find(last_foo.hid) }
+
+      let(:last_foo) { Foo.history.last }
+
+      it { is_expected.to eq last_foo }
+
+      it 'preserves associations' do
+        expect(historical_foo.bars).to eq last_foo.bars
```

Comment:
what happens if the association is set via a a dirrent attribute rather than id?

Reply:
I think that the issue is explained better in #181, but I will try to clarify at the best of my knowledge in a reasonable amount of time

Chronomodel executes `find` on Historical objects by changing the primary key in a block to `hid` with the following hack:

https://github.com/ifad/chronomodel/blob/e7c3645fdcc587070c5b98ff0e402d8258eeee03/lib/chrono_model/time_machine/history_model.rb#L29-L40

However, for this query, we need to read the `id` instead of `hid`, like it was in Rails 5.x

A Rails optimization introduced in 6.x broke this behavior: https://github.com/rails/rails/pull/36052/files#diff-3a4dbf4feb1b197bd1d507ebe3143b00615f3facd5d4469c7fa9f66f7322bcecR34

Any other attempt to read an attribute that is not `id` will result in the standard `_read_attribute` `super` call

Of course I'm scared that in some circumstances not covered by specs we need to return the `hid`, therefore I'm not willing to merge this easily. Also, this introduce a performance regression for historical models